utils/db/setupRepos.ts

```diff
@@ -3,68 +3,129 @@ import { convert } from './converter';
+		.catch(err => {
+			console.error(`[saveSetupReposInDb] Could not insert repos for user (${userId}) in the ${args.repo_owner} workspace on ${args.repo_provider}`, { pg_query: insertReposQuery }, err);
+			throw err;
+		});
+		if (reposRowCount === 0) {
+			await conn.query('ROLLBACK');
+			console.error(`[saveSetupRepos] No repositories were inserted or updated for: ${install_id}, ${repo_names}`);
+			return false;
+		}
+	
+		const repoIds = rows.map((row) => row.repo_id);
+		const insertRepoConfigSuccess = await insertRepoConfig(userId, repoIds);
+	
+		if (!insertRepoConfigSuccess) {
+			await conn.query('ROLLBACK');
+			console.error(`[saveSetupRepos] Could not insert repo configs for: ${install_id}, ${repo_names}`);
+			return false;
+		}
+		await conn.query('COMMIT');
+		console.debug(`[saveSetupRepos] setup repos info saved successfully in db`)
+		return true;
+	} catch (err) {
+		await conn.query('ROLLBACK');
+		console.error(`[saveSetupRepos] Could not save setup repos for: ${install_id}, ${repo_names}`, { pg_query: insertReposQuery }, err);
+		return false;
+	}
+};
+
+export const saveSelectedReposInDb = async (args: SetupReposArgs, userId: string): Promise<boolean> => {
+	const { repo_owner, repo_provider, repo_names, install_id } = args;
+	const insertReposQuery = `
+		INSERT INTO repos (repo_name, user_selected, repo_owner, repo_provider)
+		SELECT repo_name, ARRAY[${convert(install_id)}], ${convert(repo_owner)}, ${convert(repo_provider)}
+		FROM unnest(${convert(repo_names)}::TEXT[]) AS t(repo_name)
+		ON CONFLICT (repo_name, repo_owner, repo_provider) DO UPDATE
+			SET user_selected = CASE
+			WHEN NOT (repos.user_selected && ARRAY[${convert(install_id)}]) -- user's install_id not present
+			THEN repos.user_selected || ARRAY[${convert(install_id)}] -- append user's install_id
+			ELSE repos.user_selected -- no update needed
+			END,
+			repo_owner = ${convert(repo_owner)},
+			repo_provider = ${convert(repo_provider)}
+		RETURNING id AS repo_id;
+	`;
+	try {
+		await conn.query('BEGIN');
+		console.debug(`[saveSelectedReposInDb] insert query: `, insertReposQuery);
+		const { rowCount: reposRowCount, rows } = await conn.query(insertReposQuery)
+		.catch(err => {
+			console.error(`[saveSelectedReposInDb] Could not insert selected repos for user (${userId}) in the ${args.repo_owner} workspace on ${args.repo_provider}`, { pg_query: insertReposQuery }, err);
+			throw err;
+		});
+		if (reposRowCount === 0) {
+			await conn.query('ROLLBACK');
+			console.error(`[saveSelectedReposInDb] No repositories were inserted or updated for: ${install_id}, ${repo_names}`);
+			return false;
+		}
+	
+		const repoIds = rows.map((row) => row.repo_id);
+		const insertRepoConfigSuccess = await insertRepoConfig(userId, repoIds);
+	
+		if (!insertRepoConfigSuccess) {
+			await conn.query('ROLLBACK');
+			console.error(`[saveSelectedReposInDb] Could not insert repo configs for: ${install_id}, ${repo_names}`);
+			return false;
+		}
```

Comment:
can we remove this part?! This doesn't make any sense here since user has only selected repos here. This repo_config part already happens when DPU finally calls `setup` api after the setup